scripts/monitoring/cron-send-node-labels-status.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+""" Node label check for OpenShift V3 """
+
+# Adding the ignore because it does not like the naming of the script
+# to be different than the class name
+# pylint: disable=invalid-name
+
+# pylint: disable=wrong-import-position
+# pylint: disable=broad-except
+# pylint: disable=line-too-long
+
+import argparse
+import time
+import urllib2
+
+import logging
+logging.basicConfig(
+    format='%(asctime)s - %(relativeCreated)6d - %(levelname)-8s - %(message)s',
+)
+logger = logging.getLogger()
+logger.setLevel(logging.WARN)
+
+# Our jenkins server does not include these rpms.
+# In the future we might move this to a container where these
+# libs might exist
+#pylint: disable=import-error
+
+from openshift_tools.monitoring.ocutil import OCUtil
+from openshift_tools.monitoring.metric_sender import MetricSender
+
+ocutil = OCUtil()
+
+def runOCcmd(cmd, base_cmd='oc'):
+    """ log commands through ocutil """
+    logger.info(base_cmd + " " + cmd)
+    oc_time = time.time()
+    oc_result = ocutil.run_user_cmd(cmd, base_cmd=base_cmd, )
+    logger.info("oc command took %s seconds", str(time.time() - oc_time))
+    return oc_result
+
+def runOCcmd_yaml(cmd, base_cmd='oc'):
+    """ log commands through ocutil """
+    logger.info(base_cmd + " " + cmd)
+    ocy_time = time.time()
+    ocy_result = ocutil.run_user_cmd_yaml(cmd, base_cmd=base_cmd, )
+    logger.info("oc command took %s seconds", str(time.time() - ocy_time))
+    return ocy_result
+
+def parse_args():
+    """ parse the args from the cli """
+    parser = argparse.ArgumentParser(description='Check all the nodes label Status')
+    parser.add_argument('-v', '--verbose', action='count', default=0,
+                        help='verbosity level, specify multiple')
+    parser.add_argument('--namespace', default="default", help='service namespace')
+
+    args = parser.parse_args()
+
+    if args.verbose > 0:
+        logger.setLevel(logging.INFO)
+    if args.verbose > 1:
+        logger.setLevel(logging.DEBUG)
+
+    return args
+
+def check_lable_status():
+    """get all the info of all node """
+    result_ip = 0
+    result_status = 0
+    label_info = runOCcmd_yaml("get node ")
+    for item in label_info['items']:
+        labels = item['metadata']['labels']
+        ip_status = item['status']['addresses']
+        # if the host have less than 5 address , means node miss ip
+        if len(ip_status) < 5:
+            logger.info('this node lack of ip: %s', item['metadata']['name'])
+            result_ip = result_ip + 1
+        # if the label on the host less than 11 , means lack of label, for 3.9
+        if len(labels) < 11:
```

Comment:
would it be better to check if specific labels are available? I'm not 100% sure on which ones, but relying on a count may not work. If the team adds new labels, but important ones are still missing, then this check will still pass.

Reply:
I'd like add checks like this:
1. If it has "type=master", we should check whether the node has "node-role.kubernetes.io/master=true"
2. If it has "type=infra", we should check whether the node has "node-role.kubernetes.io/infra=true"
3. If it has "type=compute", we should check whether the node has "node-role.kubernetes.io/compute=true"